File: RootTools2/src/main/java/com/rarnu/tools/neo/xposed/ads/FuckThemeManager.java
package com.rarnu.tools.neo.xposed.ads;

import com.rarnu.tools.neo.xposed.XpUtils;
import de.robv.android.xposed.XC_MethodReplacement;
import de.robv.android.xposed.callbacks.XC_LoadPackage;

/**
 * Created by rarnu on 11/26/16.
 */
public class FuckThemeManager {

    public static void fuckThemeManager(XC_LoadPackage.LoadPackageParam loadPackageParam) {
        Class<?> clsPageItem = XpUtils.findClass(loadPackageParam.classLoader, "com.android.thememanager.model.PageItem");
        if (clsPageItem != null) {
            XpUtils.findAndHookMethod("com.android.thememanager.controller.online.PageItemViewConverter", loadPackageParam.classLoader, "buildAdView", clsPageItem, XC_MethodReplacement.returnConstant(null));
        }

    }
}

File: RootTools2/src/main/kotlin/com/rarnu/tools/neo/xposed/ads/FuckThemeManager.kt
package com.rarnu.tools.neo.xposed.ads

import android.content.Context
import com.rarnu.xfunc.*

/**
 * Created by rarnu on 11/26/16.
 */
object FuckThemeManager {

    fun fuckThemeManager(pkg: XposedPkg) {
        val clsPageItem = pkg.findClass("com.android.thememanager.model.PageItem")
        if (clsPageItem != null) {
            pkg.findClass("com.android.thememanager.controller.online.PageItemViewConverter").findMethod("buildAdView", clsPageItem).hook { replace { result = null } }
        }

        val clsHybridView = pkg.findClass("miui.hybrid.HybridView")
        if (clsHybridView != null) {
            pkg.findClass("com.android.thememanager.h5.ThemeHybridFragment\$BaseWebViewClient").findMethod("shouldInterceptRequest", clsHybridView, String::class.java).hook {
                before {
                    val url = args[1] as String
                    if (url.contains("AdCenter")) {
                        args[1] = "http://127.0.0.1/"
                    }
                }
            }
        }

        // 0.8
        pkg.findClass("com.android.thememanager.util.ApplicationHelper").apply {
            findMethod("isFreshMan").hook { replace { result = false } }
            findMethod("hasFreshManMarkRecord", Context::class.java).hook { replace { result = false } }
        }

        pkg.findClass("com.miui.systemAdSolution.landingPage.LandingPageService").findMethod("init", Context::class.java).hook { replace { result = null } }

    }
}
